### PR TITLE
Fix For Parser Sometimes Getting Thrown Off By Comments

### LINKED ISSUE #6: Lexer does not ignore comments
We need to add a case for the lexer so that when it encounters '#', it ignores the rest of the line

### PR DESCRIPTION
**Pull request by nko12**
This pull request is to fix #6, where the parser could sometimes be thrown off by comments. This also adds folders used for object files, binaries, and VS Code to the `.gitignore` file.

I did some more testing with comments and whitespace, and it seems like the parser handles things properly now. Feel free to verify this with funky files that are _still technically correct_. I don't think my parser code handles all invalid file cases smoothly, so I'll be emailing the professor to ask if he plans on using invalid test files.

### CODE REVIEW

File: Scheduler/Scanner.c [hunk 1]
```diff
@@ -92,7 +92,7 @@ void ScannerReadNextString(char** strStart, unsigned int* strLength, Scanner* sc
     *strStart = scan->fileContents + scan->currentPos;
 
     // Move the scanner forward until a whitespace character is found, or the end of the file is reached.
```
Comment (Areax) [inline Scheduler/Scanner.c:94]: The comment should now read `...until an non-alphanumeric character is found...` right?

Comment (nko12) [reply Scheduler/Scanner.c:94]: As of now, it should. But in the end, that's going to depend on the answer the professor gives me about whether or not process names can have non alphanumeric symbols in them, such as a process named "P_1$$" or something bizarre like that.

In the end, the comment is only applicable to code I maintain, so I'll fix it in some other commit or if I need to add a commit to this pull request.

**Review by pH34r-pH — APPROVED** [review]

**Review by Areax — APPROVED** [review]
